### PR TITLE
Add match visibility tests

### PR DESCRIPTION
**Pull request by cgswords**
## Description 

This adds tests to ensure visibility is correctly enforced for matching

## Test plan 

New tests

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:

### CODE REVIEW

File: external-crates/move/crates/move-compiler/tests/development/enums/matching/enum_visibility.exp [hunk 1]
```diff
@@ -0,0 +1,7 @@
+error[E04001]: restricted visibility
+   ┌─ tests/development/enums/matching/enum_visibility.move:15:13
+   │
+15 │             m::E::A(t) => t,
+   │             ^^^^^^^^^^ Invalid pattern for '0x42::m::E::A'.
+ All enums can only be matched in the module in which they are declared
```
Comment (tnowacki) [inline external-crates/move/crates/move-compiler/tests/development/enums/matching/enum_visibility.exp:6]:
```suggestion
 All enums can be matched only in the module in which they are declared
```

File: external-crates/move/crates/move-compiler/tests/development/enums/matching/struct_visibility.exp [hunk 2]
```diff
@@ -0,0 +1,7 @@
+error[E04001]: restricted visibility
+   ┌─ tests/development/enums/matching/struct_visibility.move:13:13
+   │
+13 │             m::S { t } => t,
+   │             ^^^^^^^^^^ Invalid pattern for '0x42::m::S'.
+ All struct can only be matched in the module in which they are declared
```
Comment (tnowacki) [inline external-crates/move/crates/move-compiler/tests/development/enums/matching/struct_visibility.exp:6]:
```suggestion
 All structs can be matched only in the module in which they are declared
```

**Review by tnowacki — APPROVED** [review]
Small grammar nits

File: external-crates/move/crates/move-compiler/tests/development/enums/matching/struct_visibility.exp (hunk 2 above)
Comment (cgswords) [reply external-crates/move/crates/move-compiler/tests/development/enums/matching/struct_visibility.exp:6]: "can only be X" is consistent with wording across the rest of the compiler, e.g.,

- "Invalid instantiation of '{}::{}'.\nAll structs can only be constructed in"
- "Non-u64 constants can only be used as error codes if"
- "Lambdas can only be used directly as arguments"

**Pull request merged**